### PR TITLE
refactor(elixir): Use format strings

### PR DESCRIPTION
**Pull request by jletey**
#### Description

This PR deprecates `symbol` `version` `otp_version` `style` keys in `ElixirConfig`, and adds `format` key instead.

Based off of #1063.

#### Motivation and Context

Related #1057

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):

Screenshot 2020-04-24 at 10 24 11

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by jletey** [comment]
Ah great, when I change the merge branch to `feat/format-string` it includes everyone else commits. @heyrict Could you pull master into your branch?

**Comment by heyrict** [comment]
@jletey master merged to `feat/format-string`

**Comment by filmor** [comment]
Seeing I just (hopefully) finished adding a separate Erlang/OTP module in #1129, how would you feel about dropping the OTP version from this one entirely and just making the `erlang` module show also for present `mix.config` files?

**Comment by jletey** [comment]
@filmor I'm not the creator of the `elixir` module, but this seems practical. Will implement your suggested changes :)

File: src/configs/elixir.rs [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
 
-use ansi_term::{Color, Style};
 use starship_module_config_derive::ModuleConfig;
 
 #[derive(Clone, ModuleConfig)]
 pub struct ElixirConfig<'a> {
-    pub symbol: SegmentConfig<'a>,
-    pub version: SegmentConfig<'a>,
-    pub otp_version: SegmentConfig<'a>,
-    pub style: Style,
```
Comment (andytom) [inline src/configs/elixir.rs:11]: We are keeping the `symbol` and `style` configuration options. Can you add these back please.

File: src/modules/elixir.rs [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+use regex::Regex;
 const ELIXIR_VERSION_PATTERN: &str = "\
 Erlang/OTP (?P<otp>\\d+)[^\\n]+
 
 Elixir (?P<elixir>\\d[.\\d]+).*";
 
 /// Create a module with the current Elixir version
 ///
-/// Will display the Rust version if any of the following criteria are met:
+/// Will display the Elixir version if any of the following criteria are met:
```
Comment (andytom) [inline src/modules/elixir.rs:15]: 👍 good catch

File: src/modules/elixir.rs [hunk 3]
```diff
@@ -20,73 +21,56 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
...
 #[cfg(test)]
 mod tests {
-    use super::*;
+    // use super::*;
     use crate::modules::utils::test::render_module;
     use ansi_term::Color;
     use std::fs::File;
     use std::io;
 
-    #[test]
```
Comment (andytom) [inline src/modules/elixir.rs:67]: Might be a silly question but why is this test being removed?

**Comment by heyrict** [comment]
I've updated the PR to the latest changes on feat/format-string and fixed the tests. Could you please review it again?

File: docs/config/README.md [hunk 4]
```diff
@@ -651,11 +651,22 @@ The module will be shown if any of the following conditions are met:
...
+| `style`    | `"bold purple"`                                           | The style for the module.                                       |
+| `format`   | `"via [$symbol$version \\(OTP $otp_version\\)]($style) "` | The format for the module elixir.                               |
+| `disabled` | `false`                                                   | Disables the `elixir` module.                                   |
+
+### Variables
+
+| Variable | Example | Description                          |
+|----------|---------|--------------------------------------|
+| version  | `v1.10` | The version of `elixir`              |
+| symbol   |         | Mirrors the value of option `symbol` |
```
Comment (andytom) [inline docs/config/README.md:666]: I think we need to add the `$otp_version` variable here.

Comment (heyrict) [reply docs/config/README.md:666]:
```suggestion
|otp_version|         | The otp version of `elixir`          |
| symbol    |         | Mirrors the value of option `symbol` |
```

**Comment by heyrict** [comment]
Thanks for your review :) Going to merge this now.

**Pull request merged**